drivers/soundwire/intel_init.c

```diff
@@ -134,6 +136,52 @@ sdw_intel_scan_controller(struct sdw_intel_acpi_info *info)
 	return 0;
 }
 
+#define HDA_DSP_GEN_BASE                0x0
+#define HDA_DSP_REG_ADSPIC2             (HDA_DSP_GEN_BASE + 0x10)
+#define HDA_DSP_REG_ADSPIS2             (HDA_DSP_GEN_BASE + 0x14)
+
+static void sdw_intel_enable_irq(struct sdw_intel_ctx *ctx, bool enable)
+{
+	u32 val;
+
+	val = readl(ctx->mmio_base + HDA_DSP_REG_ADSPIC2);
+
+	if (enable)
+		val |= BIT(5);
+	else
+		val &= ~BIT(5);
+
+	writel(val, ctx->mmio_base + HDA_DSP_REG_ADSPIC2);
+
+}
+
+static irqreturn_t sdw_intel_irq(int irq, void *dev_id)
+{
+	struct sdw_intel_ctx *ctx = dev_id;
+	u32 int_status;
+
+	int_status = readl(ctx->mmio_base + HDA_DSP_REG_ADSPIS2);
+	if (int_status & BIT(5)) {
+		sdw_intel_enable_irq(ctx, false);
+		return IRQ_WAKE_THREAD;
+	}
+
+	return IRQ_NONE;
+}
+
+
+static irqreturn_t sdw_intel_thread(int irq, void *dev_id)
+{
+	struct sdw_intel_ctx *ctx = dev_id;
+	struct sdw_intel_link_res *link;
+
+	list_for_each_entry(link, &ctx->link_list, list)
+		sdw_cdns_irq(irq, link->cdns);
+
+	sdw_intel_enable_irq(ctx, true);
```

Comment:
so if you don't do this disable/enable, then you need to use the repeat loop?

This looks correct, but I wonder if it has any latency impact. the repeat might be more efficient. Have you checked how many times you do the repeat typically?

Reply:
I am also thinking it. Yes, it is typically one. However, it can reach to the limitation (10) when codec try to set a lot of registers in a short time.